### PR TITLE
Added the turkish localization

### PR DESCRIPTION
**Pull request by OsmanBuyuksar**
Solution for this issue: https://github.com/goatshriek/stumpless/issues/388#issue-1971501202

Implemented turkish localization as described [here](https://github.com/goatshriek/stumpless/blob/latest/docs/localization.md)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -210,6 +210,9 @@ elseif(LOCALE MATCHES "^da[_-]")
 elseif(LOCALE MATCHES "^he[_-]")
   set(STUMPLESS_LANGUAGE "he-IL")
   set(USE_LOCALE_HE_IL TRUE)
+elseif(LOCALE MATCHES "^tr[_-]")
+  set(STUMPLESS_LANGUAGE "tr-TR")
+  set(USE_LOCALE_HE_IL TRUE)
```
Comment (goatshriek) [inline CMakeLists.txt]: This should instead be `USE_LOCALE_TR_TR`.

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thank you very much for contributing this! Just two changes to make, the first listed below. The second is to add a new section to the `.github/workflows/locale.yml` for the new translation. Commit 5529e077eb75c6e79d5634e9d7fc708828b78f89 has a good example of how to make this change if you need something to look at for comparison.

**Comment by OsmanBuyuksar** [comment]
Sry about that, i have 2 repos to keep the contributing repo clean and looks like i made some of the changes to other repo :D. Anyways i fixed locale in cmakelists and also added the tr section to locale.yml file. Let me know if there is anything else.

**Comment by goatshriek** [comment]
Looks like there's just one more thing to check - at least one line is longer than 80 characters. Once you've broken that up this should be all good to go.

**Comment by OsmanBuyuksar** [comment]
There shouldn't be any line longer than 80 characters now, is there anything else?

**Pull request merged**